pkg/engine/update.go

```diff
@@ -204,11 +204,11 @@ func Update(u UpdateInfo, ctx *Context, opts UpdateOptions, dryRun bool) (
 func RunInstallPlugins(
 	proj *workspace.Project, pwd, main string, target *deploy.Target, plugctx *plugin.Context,
 ) error {
-	_, _, err := installPlugins(proj, pwd, main, target, plugctx, true /*returnInstallErrors*/)
+	_, _, err := installPlugins(context.Background(), proj, pwd, main, target, plugctx, true /*returnInstallErrors*/)
```

Comment:
Another option here would be to use `plugctx.Request()` as the context, such that `RunInstallPlugins` cancels automatically when the plugctx is closed. I chose not to do that, because `plugctx.Request()` seems to be designed to support cancelation of outstanding RPCs to the plugin (hence the name `Request`).